server/src/logic/services/user.service.ts

```diff
@@ -1,5 +1,6 @@
 import { IDatabase } from "pg-promise";
```

Comment:
hmm, i think the register need same refactoring. this not working as before. it should return the errortype AUTH_USED_PASSWORD_EMAIL, but it returns Internal server error.

this is because you handle the user db errors inside the user.service and not simply return the promise.